examples/data-objects/inventory-app/src/inventoryList.ts

```diff
@@ -4,21 +4,42 @@
  */
 
 import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { ContainerSchema } from "fluid-framework";
 
-import { TreeDataObject, factory } from "./reactSharedTreeView.js";
+import { TreeDataObject, factory, treeDataObject } from "./reactSharedTreeView.js";
 import { type Inventory, treeConfiguration } from "./schema.js";
 
 /**
- * @internal
+ * For use with high level service-client container API's like AzureClient and OdspClient.
+ * @privateRemarks
+ * TODO: There is no service implementation agnostic client abstraction that can be referred to here (ex: shared by AzureClient and OdspClient).
+ * This makes documenting compatibility with that implicit common API difficult.
+ * It also makes writing service agnostic code at that abstraction level harder.
+ * This should be fixed.
+ *
+ * TODO:
+ * Writing an app at this abstraction level currently requires a lot of boilerplate which also requires extra dependencies.
+ * Since `@fluid-example/example-utils` doesn't provide that boilerplate and neither do the public packages, there isn't a concise way to actually use this container in this example.
+ * This should be fixed.
+ *
+ * TODO:
+ * The commonly used boilerplate for setting up a ContainerSchema based application configures the dev-tools, which would be great to include in this example,
```

Comment:
I'm not sure I understand this comment. You should be able to create the dev-tools by importing:
```typescript
import { initializeDevtools } from "@fluidframework/devtools";
```

Reply:
Or are you suggesting the data object scheme here is incompatible. That may be the case, and if so should be investigated.

Reply:
if you do that fluid layer check objects. There is apparently work planned to fix that (make dev tools not depend on exmaples)

Reply:
More generally I'd like to include the boilerplate like https://github.com/microsoft/FluidExamples/blob/625b8fdb7d2c8638cefba293e2fa8520ae8c5e84/brainstorm/src/index.tsx and everything from https://github.com/microsoft/FluidExamples/tree/625b8fdb7d2c8638cefba293e2fa8520ae8c5e84/brainstorm/src/infr

Reply:
Ugh. Okay, I understand. I _think_ you can add exceptions to the layer checker, (the dependency from devtools is a dev dependency only, so adding the import shouldn't cause any real issues). But if not, the work to clean up that dependency is scheduled for this sprint at least.

Reply:
Eventually (once this is split into its own package so others can use it), resolving this limit will be much more valuable, but for now this can just wait until the issue is resolved.